src/random-seed/random-seed.c

```diff
@@ -66,11 +94,13 @@ int main(int argc, char *argv[]) {
         if (buf_size <= POOL_SIZE_MIN)
                 buf_size = POOL_SIZE_MIN;
 
-        buf = malloc(buf_size);
-        if (!buf) {
+        entropy = (struct rand_pool_info*) malloc(sizeof(struct rand_pool_info) + buf_size);
+        if (!entropy) {
                 r = log_oom();
                 goto finish;
         }
+        entropy->buf_size = (typeof(entropy->buf_size)) buf_size;
```

Comment:
Please don't. entropy->buf_size is an int. Let's not try to hide this.
Removing the cast altogether doesn't result in a warning here, so it should just go away.